cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java

```diff
@@ -17,87 +17,75 @@
 package co.cask.cdap;
 
 import co.cask.cdap.api.app.AbstractApplication;
-import co.cask.cdap.api.mapreduce.AbstractMapReduce;
-import co.cask.cdap.api.mapreduce.MapReduceContext;
 import co.cask.cdap.api.workflow.AbstractWorkflow;
-import co.cask.cdap.internal.app.runtime.batch.WordCount;
-import org.apache.hadoop.mapreduce.Job;
+import co.cask.cdap.api.workflow.AbstractWorkflowAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.Map;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
 
 /**
  *
  */
 public class WorkflowAppWithFork extends AbstractApplication {
+  public static final String SYNCH_ON_FILE = "/tmp/WorkflowAppWithFork.done";
```

Comment:
Don't write to tmp. Takes a runtime arg and have the unit-test create a temp file (using `TemporaryFolder`) and pass in the pass through runtime arg.